buildSrc/src/main/java/tech/pegasys/internal/license/GroupedLicensesHtmlRenderer.java

```diff
@@ -0,0 +1,449 @@
+            licenses.forEach(
+                license ->
+                    licenseGroup
+                        .computeIfAbsent(license.left, k -> new ArrayList<>())
+                        .add(module)));
+    return licenseGroup;
+  }
+
+  private String getLicenseName(final String licenseName, final ModuleData moduleData) {
+    final String moduleKey = getModuleKey(moduleData);
+    if (unknownOverridesLicenses.containsKey(moduleKey)) {
+      final Map<String, String> overriddenLicenses = unknownOverridesLicenses.get(moduleKey);
+      return overriddenLicenses.getOrDefault(OVERRIDE_LICENSE, licenseName);
+    }
+    return licenseName;
+  }
+
+  private String getModuleKey(final ModuleData moduleData) {
+    return String.format(
+        "%s:%s:%s", moduleData.getGroup(), moduleData.getName(), moduleData.getVersion());
+  }
+
+  private Map<String, Map<String, String>> parseUnknownOverrides(
+      final File unknownOverridesFileName) {
+    final Map<String, Map<String, String>> unknownOverridesLicenses = new TreeMap<>();
+    try {
+      final List<String> lines =
+          java.nio.file.Files.readAllLines(
+              unknownOverridesFileName.toPath(), StandardCharsets.UTF_8);
+      lines.forEach(
+          line -> {
+            String[] columns = line.split("\\|");
+            if (columns.length > 0) {
+              final String groupName = columns[0];
+              final Map<String, String> overrideMap = new HashMap<>();
+              overrideMap.put(OVERRIDE_PROJECT_URL, columns.length > 1 ? columns[1] : null);
+              overrideMap.put(OVERRIDE_LICENSE, columns.length > 2 ? columns[2] : null);
+              overrideMap.put(OVERRIDE_LICENSE_URL, columns.length > 3 ? columns[3] : null);
+
+              unknownOverridesLicenses.put(groupName, overrideMap);
+            }
+          });
+    } catch (final IOException e) {
+      throw new UncheckedIOException(e);
+    }
+    return unknownOverridesLicenses;
+  }
+
+  private Map<String, String> parseMultiLicensesFile(final File file) {
+    final Map<String, String> multipleLicenseMap = new TreeMap<>();
+    try {
+      final List<String> lines =
+          java.nio.file.Files.readAllLines(file.toPath(), StandardCharsets.UTF_8);
+      lines.forEach(
+          line -> {
+            String[] columns = line.split("\\|");
+            if (columns.length > 0) {
+              final String groupName = columns[0];
+              final String licenseName = columns.length > 1 ? columns[1] : null;
+              multipleLicenseMap.put(groupName, licenseName);
+            }
+          });
+    } catch (final IOException e) {
+      throw new UncheckedIOException(e);
+    }
+    return multipleLicenseMap;
+  }
+
+  private void writeReport(
+      final Map<String, List<ModuleData>> licenseGroup,
+      final Map<ModuleData, Set<Pair<String, String>>> moduleLicenses) {
+    final StringBuilder html = new StringBuilder();
+    html.append("<!doctype html>\n");
```

Comment:
Don't particularly like html being directly in this class. Would be too much overhead to use a simple renderer like moustache?

Reply:
moved into its own repo.